modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/AccountResourceImpl.java

```diff
@@ -536,125 +597,68 @@ public void postOrganizationAccountsByExternalReferenceCode(
-
-		_accountEntryUserRelLocalService.setAccountEntryUserRels(
-			accountId, _getAccountUserAccountIds(account));
-
-		accountEntry = _accountEntryService.updateAccountEntry(
-			accountId,
-			_getParentAccountId(
-				account,
-				GetterUtil.getLong(
-					accountEntry.getParentAccountEntryId(),
-					AccountConstants.ACCOUNT_ENTRY_ID_DEFAULT)),
-			account.getName(), account.getDescription(),
-			_isDeleteLogo(account, null), _getDomains(account),
-			accountEntry.getEmailAddress(),
-			_getLogoBytes(account, accountEntry, false), account.getTaxId(),
-			_getStatus(account), _createServiceContext(account));
-
-		UsersAdminUtil.updateAddresses(
-			AccountEntry.class.getName(), accountEntry.getAccountEntryId(),
-			_getContactAddresses(account, null));
-		UsersAdminUtil.updateEmailAddresses(
-			AccountEntry.class.getName(), accountEntry.getAccountEntryId(),
-			_getEmailAddresses(account, null));
-		UsersAdminUtil.updatePhones(
-			AccountEntry.class.getName(), accountEntry.getAccountEntryId(),
-			_getPhones(account, null));
-		UsersAdminUtil.updateWebsites(
-			AccountEntry.class.getName(), accountEntry.getAccountEntryId(),
-			_getWebsites(account, null));
-
-		long contactId = 0;
+		com.liferay.portal.kernel.model.Organization organization =
+			_organizationService.fetchOrganizationByExternalReferenceCode(
+				externalReferenceCode, contextCompany.getCompanyId());
 
-		Contact contact = accountEntry.fetchContact();
-
-		if (contact != null) {
-			contactId = contact.getContactId();
+		if (organization == null) {
+			throw new NoSuchOrganizationException(
+				"Unable to find organization with external reference code " +
+					externalReferenceCode);
 		}
 
-		AccountContactInformation accountContactInformation =
-			account.getAccountContactInformation();
-
-		if (accountContactInformation != null) {
-			_addOrUpdateContact(
-				contactId, contextUser.getUserId(),
-				AccountEntry.class.getName(), accountEntry.getAccountEntryId(),
-				null, null, null, null, 0, 0, true, 0, 1, 1970,
-				GetterUtil.getString(accountContactInformation.getSms()),
-				GetterUtil.getString(accountContactInformation.getFacebook()),
-				GetterUtil.getString(accountContactInformation.getJabber()),
-				GetterUtil.getString(accountContactInformation.getSkype()),
-				GetterUtil.getString(accountContactInformation.getTwitter()),
-				null);
-		}
-		else {
-			_addOrUpdateContact(
-				contactId, contextUser.getUserId(),
-				AccountEntry.class.getName(), accountEntry.getAccountEntryId(),
-				null, null, null, null, 0, 0, true, 0, 1, 1970, null, null,
-				null, null, null, null);
-		}
-
-		_addAddresses(accountId, account);
-
-		return _toAccount(accountEntry);
+		postOrganizationAccountsByExternalReferenceCode(
+			organization.getOrganizationId(), externalReferenceCodes);
 	}
 
 	@Override
-	public Account putAccountByExternalReferenceCode(
-			String externalReferenceCode, Account account)
+	public Account putAccount(Long accountId, Account account)
```

Comment:
for the put we cannot call the "post" (in case of addAccountEntry) or the patch(in case of update)? Because I don't like to have all the code after the if also here because otherwise we are missing something
example: in the patch there is
```accountEntry = _accountEntryService.updateExternalReferenceCode(
			accountId,
			GetterUtil.getString(
				account.getExternalReferenceCode(),
				accountEntry.getExternalReferenceCode()));

		_addAddresses(accountId, account);

		_accountEntryLocalService.updateDefaultBillingAddressId(
			accountId,
			_getDefaultBillingAddressId(
				account, accountEntry.getDefaultBillingAddressId()));

		_accountEntryLocalService.updateDefaultShippingAddressId(
			accountId,
			_getDefaultShippingAddressId(
				account, accountEntry.getDefaultShippingAddressId()));

		long[] organizationIds = _getOrganizationIds(account);

		if (organizationIds != null) {
			_accountEntryOrganizationRelLocalService.
				setAccountEntryOrganizationRels(accountId, organizationIds);
		}

		UserAccount[] userAccounts = account.getAccountUserAccounts();

		if (userAccounts != null) {
			_accountEntryUserRelLocalService.setAccountEntryUserRels(
				accountId,
				transformToLongArray(
					Arrays.asList(userAccounts), UserAccount::getId));
		}

		AccountContactInformation accountContactInformation =
			account.getAccountContactInformation();

		if (accountContactInformation != null) {
		....
```

and we don't have the same here. I would like to have this logic only in one place (because the add done by post or put should do exactly the same thing), the update from patch or put should do the same (or better, in this case is a little different because in the patch you should keep previous values, starting from the put you should reset to the defaults if not passed)

Reply:
@smottal I refactored the patch/put stuff based on your thoughts. I think it is way better now.